src/components/App/App.js

```diff
@@ -44,28 +96,125 @@ function App() {
+      });
+  }
+
+  function handleRegistration(data) {
+    mainApi
+      .signup(data.email, data.password, data.username)
+      .then((res) => {
+        setRegisterPopUpOpen(false);
+        setRegistretionSuccsess(true);
+      })
+      .catch((err) => setRegisterErrorMsg(err.message));
+  }
+
+  function handleLogin(data) {
+    mainApi.signin(data.email, data.password).then((res) => {
+      setToken(res.token);
+      localStorage.setItem("jwt", res.token);
+      setIsLogedIn(true);
+      closeAllPopUps();
+    });
+  }
+
+  function handleLogOut() {
+    localStorage.removeItem("jwt");
+    setIsLogedIn(false);
+  }
+
+  function handleCardSave(card) {
+    const sourceChecker = card.source.id ? card.source.id : "no source";
+    const formatCard = {
+      keyword: searchKeyWord,
+      title: card.title,
+      text: card.content,
+      date: card.publishedAt,
+      source: sourceChecker,
+      link: card.url,
+      image: card.urlToImage,
+    };
+    mainApi.saveCard(token, formatCard).then(() => {
+      mainApi.getUserArticlesList(token).then((res) => {
+        localStorage.setItem("userCards", JSON.stringify(res));
+        setUserCard(res);
+      });
+    });
+  }
+
+  function handleCardDelete(card) {
+    const cardFilter = userCard.filter(
+      (userCard) => userCard.link === card.url
+    );
+    const cardId = cardFilter[0]._id;
+    mainApi.deleteCard(token, cardId).then((res) => {
+      mainApi.getUserArticlesList(token).then((res) => {
+        localStorage.setItem("userCards", JSON.stringify(res));
+        setUserCard(res);
+      });
+    });
+  }
+
+  function getUserArticlesList() {
+    mainApi.getUserArticlesList(token).then((res) => {
+      return res;
+    });
+  }
+
   return (
     <div className="page">
       <div className="page__container">
         <Routes>
           <Route
             path="/saved-news"
             element={
-              <>
-                <Header isLogedIn={isLogedIn} darMode={true} />
-                <SavedNews />
-                <Footer />
-              </>
+              <ProtectedRoute isLogedIn={isLogedIn}>
+                <CurrentUserContext.Provider value={currentUser}>
+                  <UserSavedCardsContext.Provider value={userCard}>
+                    <Header
+                      isLogedIn={isLogedIn}
+                      darMode={true}
+                      onLogOut={handleLogOut}
+                    />
+                    <SavedNews
+                      userCards={userCard}
+                      isLogedIn={isLogedIn}
+                      handleCardDelete={handleCardDelete}
+                    />
+                    <Footer />
+                  </UserSavedCardsContext.Provider>
+                </CurrentUserContext.Provider>
+              </ProtectedRoute>
             }
           />
           <Route
             path="/"
             element={
-              <>
+              <CurrentUserContext.Provider value={currentUser}>
```

Comment:
`<CurrentUserContext.Provider value={currentUser}>` should be set only once to wrap the whole app. You don't need to put it in 2-3 different places